### ISSUE TITLE
`pack build` should validate that buildpacks provided with `--buildpack` have a matching stack ID in `buildpack.toml`

### ISSUE THREAD

**Issue by ekcasey** [buildpack/pack#133]
**Given** I have a builder `my-builder` with stack ID `some.stack` and a buildpack at path `path/to/buildpack` with the following stack info in the `buildpack.toml`
```
[[stacks]]
  id = "other.stack"
```
**When** I run `pack build my-image --builder my-builder --buildpack path/to/buildpack`
**Then** I see an error informing me that the buildpack cannot be used with this builder b/c the buildpack does not support stack `some.stack`

**Given** I have a builder `my-builder` with stack ID `some.stack` and a buildpack at path `path/to/buildpack` with the following stack info in the `buildpack.toml`
```
[[stacks]]
  id = "other.stack"
[[stacks]]
  id = "some.stack"
```
**When** I run `pack build my-image --builder my-builder --buildpack path/to/buildpack`
**Then** It succeeds

**Comment by mgibson1121** [comment]
This story passes acceptance

**Comment by ekcasey** [comment]
fixed on `master`, will ship in `v0.2.0`
